pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -816,6 +795,44 @@ func (s *DiscoveryServer) removeEdsCon(clusterName string, node string) {
 	}
 }
 
+func (s *DiscoveryServer) updateEdsClients(added sets.Set, removed sets.Set, connection *XdsConnection) {
+	edsClusterMutex.Lock()
+	defer edsClusterMutex.Unlock()
+
+	for rc := range removed {
+		c := edsClusters[rc]
+		if c == nil {
+			adsLog.Warnf("EDS: Missing cluster: %s", rc)
+			return
```

Comment:
continue

Reply:
just find that this is duplicate with `removeEdsCon`

Reply:
great catch. Changed to continue. It is not duplicate with `removeEdsCon` because, `removeEdsCon` - takes lock for every connection + some extra locks. This one reuses the locks taken above